src/cpp/utils/time_t_helpers.hpp

```diff
@@ -40,8 +40,12 @@ static void current_time_since_unix_epoch(
 {
     using namespace std::chrono;
 
+    static const auto init_time_since_epoch = system_clock::now().time_since_epoch();
+    static const auto init_steady_time = steady_clock::now();
+
     // Get time since epoch
-    auto t_since_epoch = system_clock::now().time_since_epoch();
+    auto t_elapsed = steady_clock::now() - init_steady_time;
+    auto t_since_epoch = init_time_since_epoch + t_elapsed;
```

Comment:
I'm not sure I understand this, we may have intended to write
`static const auto init_steady_time = steady_clock::now().time_since_epoch();`.
But even in that case, I dont fully understand how we are protecting against adjustments in the `system_clock` by adding up the elapsed time since the last reboot (the most common return across platforms for the steady_clock's `time_since_epoch()`).

Other than that, LGTM

Reply:
We protect against adjustments in the `system_clock` by doing a single call to `system_clock::now()` the first time `current_time_since_unix_epoch` is called. We then return that initial reference plus the amount of `steady_clock` time since the first time the method was called.

The idea is to make `t_since_epoch` be the ``

Reply:
I overlooked the `static`. Thanks !